packages/signers/signers-common/src/v4.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  type V14,
+  type V15,
+  compact,
+  enhanceEncoder,
+  u8,
+} from "@polkadot-api/substrate-bindings"
+import { mergeUint8 } from "@polkadot-api/utils"
+import { getLookupFn, LookupEntry } from "@polkadot-api/metadata-builders"
+
+const versionCodec = enhanceEncoder(
+  u8.enc,
+  (value: { signed: boolean; version: number }) =>
+    (+!!value.signed << 7) | value.version,
+)
+
+export const enum SignerType {
+  Polkadot,
+  Ethereum,
+}
+const unkownSignerType = () => new Error("Unkown signer")
+const getSignerType = (metadata: V14 | V15): [SignerType, [] | [number]] => {
+  const { extrinsic } = metadata
+  const getLookup = getLookupFn(metadata)
+  let address: LookupEntry
+  let signature: LookupEntry
+  if ("address" in extrinsic) {
+    address = getLookup(extrinsic.address)
+    signature = getLookup(extrinsic.signature)
+  } else {
+    const extLkup = getLookup(extrinsic.type)
+    if (extLkup.type !== "struct") throw unkownSignerType()
+    address = extLkup.value["Address"]
+    signature = extLkup.value["Signature"]
```

Comment:
This won't work. The types are in the `params` section of the metadata, and the lookupFn does not help since it does not expose the `params`, raw metadata has to be used instead. Implementation example: https://github.com/polkadot-api/polkadot-api/blob/8325515664dd42220992622219a29ca3ba81ce75/packages/tx-utils/src/decode-extrinsic.ts#L47-L53